firefox_desktop/views/desktop_daily_retention.view.lkml

```diff
@@ -0,0 +1,355 @@
+    hidden: yes
+  }
+
+  measure: num_clients_dau_active_atleastonce_in_last_7_days_measure {
+    label: "clients - DAU at least once in last 7 days"
+    type: sum
+    sql:  ${TABLE}.num_clients_dau_active_atleastonce_in_last_7_days ;;
+    hidden: yes
+  }
+
+  measure: num_clients_active_atleastonce_in_last_28_days_measure {
+    label: "clients - sent ping at least once in last 28 days"
+    type: sum
+    sql:  ${TABLE}.num_clients_active_atleastonce_in_last_28_days ;;
+    hidden: yes
+  }
+
+  measure: num_clients_dau_active_atleastonce_in_last_28_days_measure {
+    label: "clients - DAU at least once in last 28 days"
+    type: sum
+    sql:  ${TABLE}.num_clients_dau_active_atleastonce_in_last_28_days ;;
+    hidden: yes
+  }
+
+  measure: num_clients_active_in_period {
+    label: "clients - sent a ping during the retention period "
+    type: sum
+    sql:  CASE WHEN {% parameter retention_period_picker %} = 1 THEN ${TABLE}.num_clients_active_on_day WHEN {% parameter retention_period_picker %} = 7 THEN ${TABLE}.num_clients_active_atleastonce_in_last_7_days WHEN {% parameter retention_period_picker %} = 28 THEN ${TABLE}.num_clients_active_atleastonce_in_last_28_days ELSE 0 END;;
+  }
+
+  measure: num_clients_dau_in_period {
+    label: "clients - qualified as DAU during the retention period "
+    type: sum
+    sql:  CASE WHEN {% parameter retention_period_picker %} = 1 THEN ${TABLE}.num_clients_dau_on_day WHEN {% parameter retention_period_picker %} = 7 THEN ${TABLE}.num_clients_dau_active_atleastonce_in_last_7_days WHEN {% parameter retention_period_picker %} = 28 THEN ${TABLE}.num_clients_dau_active_atleastonce_in_last_28_days ELSE 0 END;;
+  }
+
+  measure: retention_rate_measure{
+    label: "retention rate"
+    type: number
+    sql: SAFE_DIVIDE(${num_clients_active_in_period}, ${cohort_measure}) ;;
+    value_format: "0.00%"
+  }
+
+  measure: retention_rate_dau_measure{
+    label: "retained as dau rate"
+    type: number
+    sql: SAFE_DIVIDE(${num_clients_dau_in_period}, ${cohort_measure}) ;;
+    value_format: "0.00%"
+  }
+
+# Retention Dimensions
+
+  parameter: retention_period_picker {
+    description: "Which Retention Period to Look Over"
+    type: unquoted
+    allowed_value: {
+      label: "Daily"
+      value: "1"
+    }
+    allowed_value: {
+      label: "Weekly"
+      value: "7"
+    }
+    allowed_value: {
+      label: "Monthly"
+      value: "28"
+    }
+  }
+
+  dimension: retention_period {
+    label: "Retention Period"
+    type:  number
+    description: "Only Works if Selection is Made in Retention Period Picker"
+    sql: IF(MOD(DATE_DIFF(${TABLE}.submission_date, ${TABLE}.first_seen_date, DAY) + 1, {% parameter retention_period_picker %}) = 0, SAFE_DIVIDE(DATE_DIFF(${TABLE}.submission_date, ${TABLE}.first_seen_date, DAY) + 1, {% parameter retention_period_picker %}), NULL) ;;
+  }
+
+
+# Group Dimensions
+  dimension: attribution_campaign {
+    label: "Campaign"
```

Comment:
```suggestion
    label: "Campaign"
    description: "The campaign the user was acquired from. This is pulled using last non-direct click attribution from Google Analytics, and is only available for Windows. For details, see https://support.google.com/analytics/answer/1662518"
```

Something like this might be useful in giving people a sense of what this is